test/WebJobs.Script.Tests/Description/FunctionAppValidationServiceTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Host;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class FunctionAppValidationServiceTests
+    {
+        private readonly ILogger<FunctionAppValidationService> _testLogger;
+        private readonly Mock<IOptions<ScriptJobHostOptions>> _scriptOptionsMock;
+        private readonly ScriptJobHostOptions _scriptJobHostOptions;
+        private readonly TestLoggerProvider _testLoggerProvider;
+        private readonly TimeSpan _initializationDelay = TimeSpan.FromSeconds(11);
+
+        public FunctionAppValidationServiceTests()
+        {
+            _scriptOptionsMock = new Mock<IOptions<ScriptJobHostOptions>>();
+
+            _scriptJobHostOptions = new ScriptJobHostOptions
+            {
+                RootScriptPath = "test-root-path",
+                IsDefaultHostConfig = false
+            };
+
+            _scriptOptionsMock.Setup(o => o.Value).Returns(_scriptJobHostOptions);
+
+            _testLoggerProvider = new TestLoggerProvider();
+            LoggerFactory factory = new LoggerFactory();
+            factory.AddProvider(_testLoggerProvider);
+            _testLogger = factory.CreateLogger<FunctionAppValidationService>();
+        }
+
+        [Fact]
+        public async Task StartAsync_NotDotnetIsolatedApp_DoesNotLogError()
+        {
+            _testLoggerProvider.ClearAllLogMessages();
+
+            var service = new FunctionAppValidationService(
+                _testLogger,
+                _scriptOptionsMock.Object,
+                new TestEnvironment());
+
+            // Act
+            await service.StartAsync(CancellationToken.None);
```

Comment:
I think a more valuable test would be to add an E2E test which simulates specialization. Then you can remove this test class altogether (I am not convinced this test class will add any additional value after adding the above-mentioned test - I could be wrong)

Reply:
Added the test. I think unit tests are good too along with E2E tests.